cvat/apps/engine/models.py

```diff
@@ -535,3 +536,71 @@ class Comment(models.Model):
     message = models.TextField(default='')
     created_date = models.DateTimeField(auto_now_add=True)
     updated_date = models.DateTimeField(auto_now=True)
+
+class CloudProviderChoice(str, Enum):
+    AWS_S3 = 'AWS_S3_BUCKET'
+    AZURE_CONTAINER = 'AZURE_CONTAINER'
+    GOOGLE_DRIVE = 'GOOGLE_DRIVE'
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+    @classmethod
+    def list(cls):
+        return list(map(lambda x: x.value, cls))
+
+    def __str__(self):
+        return self.value
+
+class CredentialsTypeChoice(str, Enum):
+    # ignore bandit issues because false positives
+    TEMP_KEY_SECRET_KEY_TOKEN_PAIR = 'TEMP_KEY_SECRET_KEY_TOKEN_PAIR' # nosec
+    ACCOUNT_NAME_TOKEN_PAIR = 'ACCOUNT_NAME_TOKEN_PAIR' # nosec
+    ANONYMOUS_ACCESS = 'ANONYMOUS_ACCESS'
+
+    @classmethod
+    def choices(cls):
+        return tuple((x.value, x.name) for x in cls)
+
+    @classmethod
+    def list(cls):
+        return list(map(lambda x: x.value, cls))
+
+    def __str__(self):
+        return self.value
+
+class CloudStorage(models.Model):
+    provider_type = models.CharField(max_length=20, choices=CloudProviderChoice.choices())
```

Comment:
Do we need a name for cloudstorage (with https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.Field.unique True)? In some cases resource isn't the best identifier and you want to provide an alternative name.

Reply:
AWS bucket name and Azure container name are unique names. Why we need an alternative name?

Reply:
Because they can be non-user friendly or too long. Thus you want to have an alias which easy to remember and understand.